cli/commons/src/Microsoft.Kiota.Cli.Commons/IO/TableOutputFormatter.cs

```diff
@@ -0,0 +1,168 @@
+        var firstElement = GetFirstElement(root);
+
+        IEnumerable<string> propertyNames = GetPropertyNames(firstElement);
+        var table = new Table();
+        table.Expand();
+
+        foreach (var propertyName in propertyNames)
+        {
+            table.AddColumn(propertyName, column =>
+            {
+                if (firstElement.ValueKind == JsonValueKind.Object)
+                {
+                    var hasProp = firstElement.TryGetProperty(propertyName, out var property);
+                    if (property.ValueKind == JsonValueKind.Number)
+                        column.RightAligned().PadLeft(10);
+                }
+            });
+        }
+
+        if (root.ValueKind == JsonValueKind.Array)
+        {
+            foreach (var row in root.EnumerateArray())
+            {
+                var rowCols = GetRowColumns(propertyNames, row);
+                table.AddRow(rowCols);
+            }
+        }
+        else if (root.ValueKind == JsonValueKind.Object)
+        {
+            var rowCols = GetRowColumns(propertyNames, root);
+            table.AddRow(rowCols);
+        }
+        else
+        {
+            table.AddRow(GetPropertyValue(root));
+        }
+
+        return table;
+    }
+
+    private static JsonElement GetRootElement(JsonElement input) {
+        var root = input;
+        if (root.ValueKind == JsonValueKind.Object && root.TryGetProperty("value", out var value))
+            root = value;
+        
+        return root;
+    }
+
+    private static JsonElement GetFirstElement(JsonElement root) {
+        var firstElement = root;
+        if (root.ValueKind == JsonValueKind.Array && root.GetArrayLength() > 0)
+        {
+            var enumerated = root.EnumerateArray();
+            firstElement = enumerated.FirstOrDefault();
+        }
+        
+        return firstElement;
+    }
+
+    private static IEnumerable<string> GetPropertyNames(JsonElement firstElement) {
+        IEnumerable<string> propertyNames;
+        if (firstElement.ValueKind != JsonValueKind.Object)
+        {
+            propertyNames = new List<string> { "Value" };
+        }
+        else
+        {
+            var restrictedValueKinds = new JsonValueKind[] {
+                    JsonValueKind.Array,
+                    JsonValueKind.Object
+                };
+            var objectEnumerator = firstElement.EnumerateObject();
+            var buffer = new List<string>();
+            foreach (var property in objectEnumerator)
+            {
+                if (restrictedValueKinds.Contains(property.Value.ValueKind)) {
+                    continue;
+                }
+
+                buffer.Add(property.Name);
+            }
+            propertyNames = buffer;
+        }
+        
+        return propertyNames;
+    }
+
+    private static IEnumerable<IRenderable> GetRowColumns(IEnumerable<string> propertyNames, JsonElement row)
+    {
+        return propertyNames.Select(p =>
```

Comment:
this is potentially a long call, maybe we should throw if cancelled here, but as selects run over vectors (SIMD), I'm not sure whether we're supposed to do it or not, can you do a bit of research on that please?

Reply:
Referenced in new issue https://github.com/microsoft/kiota/issues/1366